src/pages/redirect-protection/index.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { mount, html } from 'hybrids';
+import '/ui/index.js';
+import { themeToggle } from '/ui/theme.js';
+
+function goBack() {
+  if (window.history.length > 1) {
+    window.history.back();
+  } else {
+    chrome.tabs.getCurrent((tab) => {
+      if (tab) {
+        chrome.tabs.remove(tab.id);
+      }
+    });
+  }
+}
+
+async function allow(host) {
+  if (!host.targetUrl) return;
+
+  try {
+    const response = await chrome.runtime.sendMessage({
+      action: 'allowRedirect',
+      url: host.targetUrl,
+    });
+
+    if (response?.success) {
+      location.replace(host.targetUrl);
+    }
+  } catch (error) {
+    console.error('[redirect-protection] Failed to allow redirect:', error);
+  }
+}
+
+async function alwaysAllow(host) {
+  if (!host.hostname) return;
+
+  try {
+    const response = await chrome.runtime.sendMessage({
+      action: 'disableRedirectProtection',
+      hostname: host.hostname,
+    });
+
+    if (response?.success) {
+      location.replace(host.targetUrl);
+    }
+  } catch (error) {
+    console.error('[redirect-protection] Failed to disable protection:', error);
+  }
+}
+
+const App = {
```

Comment:
We don't call main component `App`. Usually, this is a name of the module, or you can inline it at the last line of file.